### ISSUE TITLE
Media files are not generated on multisite setup in 3.2.0

### ISSUE THREAD

**Issue by helllicht** [getkirby/kirby#1901]
**Describe the bug**
Since upgrading to 3.2.0, Kirby is not generating the media in the `/media/` directory for pages anymore, or at least not completely. In some cases it creates empty folders (with hashed names, as expected) but doesn't create any images.

The panel files itself are still working for the most part, although it doesn't create the necessary `index.js`and `index.css` files inside the `plugins` directory (panel/plugins). Seems like the issues are related.

We're using a multisite setup (different content folders for different domains), so our paths for the project are not `/media/pages/…` but `/media/client-website.com/pages/…`

**To Reproduce**
1. Install 3.2.0
2. Use Multilang setup (inside of `index.php`, see below)
3. Delete media folder contents
4. Visit frontend of the website and notice missing images or to the panel and see missing fields

Here's the multisite setup we're using inside of `index.php`:

```
<?php

require 'kirby/bootstrap.php';

$domains = array('some-domain.com, 'some-other-domain.com');
$domain  = url::host();

if (in_array($domain, $domains)) {
  $kirby = new Kirby([
    'roots' => [
      'index'   => __DIR__,
      'site'    => __DIR__ . '/site/',
      'content' => __DIR__ . '/content/' . $domain,
      'media'   => __DIR__ . '/media/' . $domain,
      'cache'   => __DIR__ . '/site/cache/' . $domain,
      'accounts' => __DIR__ . '/site/accounts/' . $domain,
    ],
    'urls' => [
      'media'   => url() . '/media/' . $domain,
    ]
  ]);

  echo $kirby->render();

} else {
  echo (new Kirby)->render();
}
```

**Expected behavior**
Kirby should create all necessary thumbnails and panel files as as was before :)

**Kirby Version**
3.2.0

**Comment by LeBenLeBen** [comment]
Thanks for posting this, I've been looking for an hour to understand what was going on with `media/plugins/index.js` and `media/plugins/index.css`. The server return a 404 error when it's loaded through the panel page, but when you access it directly you can actually see the content of file with the right mime-type, although the http code is still 404 for some reason.

**Comment by lukasbestle** [comment]
✅

**Comment by LeBenLeBen** [comment]
Is it supposed to generate the files in the public folder again now? (EDIT: I guess yes since creating a fake file in there fix it). I updated to 3.2.3 and I still have the bug but only on the remote server. Plugins' index.css and index.js files doesn't get generated neither in development or on the remote server, but locally it works fine.

On the remote I still have the same issue where the http code returned is 404 but accessing the URL of the file directly properly returns the file content.

I can give you access to my instance if that can help you debug it, feel free to ask me on Slack.

@helllicht Did the fix released in 3.2.3 worked for you?

EDIT:
* I've dumped the value of the `$uri` @bastianallgeier changed in `src/Http/Uri.php` and it's slighty different between local and remote; local doesn't have a trailing slash whereas remote does have it.
* There's another difference between the two: remote has SSL.
